bundles/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/bus/KNXBinding.java

```diff
@@ -108,7 +108,7 @@ public void removeKNXTypeMapper(KNXTypeMapper typeMapper) {
     @Override
     protected void internalReceiveCommand(String itemName, Command command) {
         logger.trace("Received command (item='{}', command='{}')", itemName, command.toString());
-        if (!isEcho(itemName, command)) {
+        if (!KNXConnection.getCompatibilityOH2() && !isEcho(itemName, command)) {
```

Comment:
> internalReceiveCommand is called AND directly after internalReceiveUpdate is called

If autoupdate is not suppressed, this is normal behavior for both OH1 and OH2.  I don't see how many other 1.x bindings would not be failing if this weren't true.  Does this binding attempt to suppress autoupdate logic but fails to do so on OH2?

> I believe there is not the risk of a memory leak at this point. Am I wrong?

Before this PR, the code was `if (B)`, so `B` is evaluated every time the `if` statement is reached.  In the code `if (A && B)`, if `A` is `false`, `B` is never evaluated, so no code removes the ignore event from the list, leading to a memory leak.